### ISSUE TITLE
Suggestion: Toggle Troop Types

### ISSUE THREAD

**Issue by raechu** [TLDRMissions/TLDRMissions#409]
**TLDR Version of your idea**
Adding an option to select which type of troop to prefer, or turn off simulating with one of the troop types entirely, would significantly reduce the number of simulations required.

**Slightly longer version**
Troops aren't really all created equally. Maldraxxus's ranged troops are amazing dps, while their melee troops are pretty garbage. Ardenweald's melee troops are better than a lot of actual followers, and their healer troops just aren't worth using at all. If I could tell the addon to not bother simulating missions with the troop type that is weak, or to do all the sims with the better troop type first before attempting to mix and match, it would reduce the average amount of simulations required per mission by quite a lot, with only a tiny % reduction in my ability to complete a mission.

**Additional context**
Let's be honest the vast majority of us have our mission table alts sit in Ardenweald because the Trapper troop is so OP. I came up with this idea when i noticed that the addon was sending out missions with a lot of the Grovetender troops instead. I feel like the addon is wasting time calculating missions with the Grovetender troop when the Trapper is so so so much stronger. So I thought if I had the option to disable the Grovetender, I could speed up the calculating by quite a lot! This could also reduce the impact of those difficult missions where I hit the 30,000 simulation max and have to skip them.

**Comment by teelolws** [comment]
> This could also reduce the impact of those difficult missions where I hit the 30,000 simulation max and have to skip them.

For this, I've been working on a different approach for this. If you look in the ResultCache folder in the addon, you'll see 100-ish files for all the missions I've spent computing time on so far. Its a list of which followers cannot solo that mission, no matter what. I haven't got a file for all the missions yet, as it takes a lot of computer power and time to gather them, but I'm slowly working on it.

As for vetoing a troop, I'll consider it in the future.

**Comment by teelolws** [comment]
I've been watching the logs and setups and have decided not to implement your troops filter request. While Ardenweald Trappers are the best troop for a majority of missions, there are plenty of missions where Grovetenders are actually better, and some missions need a mixture of both troops to succeed.

If you're so concerned about how long it takes to Simulate, and don't mind the 4 extra anima cost for including troops, just leave the addon on the default setting of requiring 4 troops for every mission.
